### ISSUE TITLE
Can't construct parent environment of CCWeights namespace

### ISSUE THREAD

**Issue by krlmlr** [cynkra/constructive#365]
Haven't tested other packages.

``` r
requireNamespace("CCWeights")
#> Loading required namespace: CCWeights
constructive::construct(getNamespace("CCWeights"), constructive::opts_environment(predefine = TRUE))
#> asNamespace("CCWeights")
constructive::construct(parent.env(getNamespace("CCWeights")), constructive::opts_environment(predefine = TRUE))
#> Error in `constructive::construct()`:
#> ! {constructive} could not build the requested code.
#> Caused by error in `deparse_call()`:
#> ! `call` must only be made of symbols and syntactic literals
#> Caused by error in `deparse_call_impl()`:
#> ! found element of type 'list' and length '0':
#> list()
```

<sup>Created on 2024-04-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Relevant for collector/duckplyr: I want to turn the collected env into a reprex, and this is what collector seems to capture.

**Comment by moodymudskipper** [comment]
I want to understand and solve this bug but this instance would go out of the window if I tackled #75 , for your use case #75 would be better right ? no need to reproduce the code of all imports.

**Comment by krlmlr** [comment]
I think so, yes.

Antoine Fabri ***@***.***> schrieb am Do. 18. Apr. 2024 um
12:50:

**Comment by krlmlr** [comment]
Worth filing upstream?

Can/should we improve our error message?

For collector, we can work around, right?

Antoine Fabri ***@***.***> schrieb am Do. 18. Apr. 2024 um
13:30:

**Comment by moodymudskipper** [comment]
We have another issue for the object `debug_names`, for a different PR

**Comment by moodymudskipper** [comment]
We have another issue when parsing the output, it boils down to: 

``` r
constructive::construct(c("\\c" = 1))
#> Error in `constructive::construct()`:
#> ! The code built by {constructive} could not be parsed.
#> Caused by error:
#> ! '\c' is an unrecognized escape in character string starting "`\c"
```

deparse() does no better than constructive here

**Comment by moodymudskipper** [comment]
The above is fixed but I still can't `construct(tools::Rd2HTML)`, though a copy with the same body works fine

**Comment by moodymudskipper** [comment]
The names in deparse_call_impl were receiving a different treatment, now fixed but we still have the dollar to fix, the code chokes on

```
boxTag$children[[2]]$attribs$data-for <-
```

**Comment by moodymudskipper** [comment]
now fixed as well, unfortunately waldo compares named environments by reference so it sees a difference. This is purposeful in waldo so is unlikely to change but I opened a ticket : https://github.com/r-lib/waldo/issues/161 .
Ideally constructive itself should contain some sort of waldo rewrite but it's a lot of work for limited gain.
In any case after #75 no difference will be shown for the "imports" env
